cpp/src/arrow/flight/sql/client.cc

```diff
@@ -369,8 +369,9 @@ arrow::Result<int64_t> PreparedStatement::ExecuteUpdate() {
   } else {
     const std::shared_ptr<Schema> schema = arrow::schema({});
     ARROW_RETURN_NOT_OK(client_->DoPut(options_, descriptor, schema, &writer, &reader));
+    const auto& columns = std::vector<std::shared_ptr<Array>>{};
```

Comment:
nit (assuming `arrow/type_fwd.h` is included):
```suggestion
    ArrayVector columns;
```